### DISCUSSION TITLE
Window Rule for blocking idle requests when full screen

### DISCUSSION THREAD

**Discussion by squishram** [YaLTeR/niri#742] [Ideas]
Currently, if I'm watching a video on fullscreen in Niri, swayidle will trigger as normal. It would be nice to be able to set a window rule that inhibits swayidle and any other idle/ screensaver processes when the window is in fullscreen mode. As it is, I need to regularly jiggle the mouse to stop this happening. Sway and Hyprland seem to deal with this with their implementations of window rules.

Thank you for your work on this project by the way! This is a gorgeous compositor with a really intuitive workflow.

**Comment by YaLTeR** [comment]
What are you watching a video in? Normally the application requests the session idle to be inhibited. Though, I agree such a window rule would make sense.

> Thank you for your work on this project by the way! This is a gorgeous compositor with a really intuitive workflow.

Thanks!

**Reply by squishram** [reply]
Behaviour does vary with the application/ video player being used (mpv/ YouTube seem to be fine) -- some of them send a request, some of them don't. It's not exactly a game-breaking problem, but I suppose it's always nice for these behaviours to be in the user's hands in cases of service developer oversight. It's a shame that swayidle itself doesn't deal with this, but I imagine it's tricker if not impossible for the idle daemon to handle this kind of thing.

**Comment by txkyel** [comment]
I'd like for this feature too. In hyprland I would use the window rule [`idleinhibit`](https://wiki.hyprland.org/Configuring/Window-Rules/#dynamic-rules) matching any fullscreened application `windowrule = idleinhibit fullscreen, fullscreen:1`  which would allow me to fullscreen a terminal and prevent idling.

**Reply by LBY-L** [reply]
Well, also when i fullscreen things like my terminal or some applications that don't request the session idle to be inhibited, it just become idle and it's annoying, also idk if it's a bug of some kind, but when i use the idle inhibitor of waybar in a fullscreen state with that type of applications, the idle inhibitor just get ignored, and then the idle comes and makes their job.

https://github.com/user-attachments/assets/1464c179-4139-49d2-b6e8-012d7cb02669

**Comment by rachitvrma** [comment]
Zathura pdf reader doesn't send the idle inhibit signal as well.

**Comment by MrDowntempo** [comment]
oooh, this is a feature I was looking for. Subscribing to hopefully be informed if and when it gets implemented so I don't have to use a 3rd party tool like caffeine or run some extra systemd idle inhibitor service.

**Comment by Disonantemus** [comment]
Sway has: [inhibit_idle focus | fullscreen | open | none | visible](https://man.archlinux.org/man/sway.5.en), and you can do things like:

    for_window [app_id="chromium"] inhibit_idle fullscreen
    for_window [app_id=".*"] inhibit_idle fullscreen
    for_window [shell=".*"] inhibit_idle fullscreen
    for_window [class=".*"] inhibit_idle fullscreen

- Is possible to do this with **Niri**?

---

### Some workarounds:

- https://github.com/rafaelrc7/wayland-pipewire-idle-inhibit
- https://github.com/ATTron/swaddle
- https://github.com/bdwalton/inhibit-bridge
- https://github.com/ErikReider/SwayAudioIdleInhibit
- https://github.com/codevardhan/caffeine-applet
- https://github.com/loops/idlehack

**Comment by griccardos** [comment]
I have given the implementation a try.
You can test on https://github.com/griccardos/niri/tree/inhibit_idle
I have opened a pull request #2373, see there for more details

I have added the option to inhibit on both `fullscreen` and `always` .

During implementing this feature I actually ended up also writing a tray inhibitor, which prevents inhibiting for a period of time, and works in niri and other window managers. Perhaps another option in addition to the above workarounds
https://github.com/griccardos/idlehands
